server/player/src/packet_handlers/placement.rs

```diff
@@ -24,66 +24,102 @@ pub fn handle_player_block_placement(
     packet_buffers
         .received::<PlayerBlockPlacement>()
         .for_each_valid(world, |world, (player, packet)| {
-            // TODO: handle slabs, blocks with directions, etc.
-            let gamemode = *world.get::<Gamemode>(player);
-            let inventory = world.get::<Inventory>(player);
-
-            let item = match inventory.item_in_main_hand(player, world) {
-                Some(item) => item,
-                None => return, // No block to place
-            };
-
-            drop(inventory);
-
-            let block = match item.ty.to_block() {
-                Some(block) => block,
-                None => return, // Item is not a block
-            };
-
-            let placed_on = match game.block_at(packet.location) {
+            let target_block = match game.block_at(packet.location) {
                 Some(block) => block,
                 None => {
-                    game.disconnect(player, world, "attempted to place block in unloaded chunk");
+                    game.disconnect(
+                        player,
+                        world,
+                        "Attempted to interact with block in unloaded chunk",
+                    );
                     return;
                 }
             };
 
-            // TODO: waterlogged blocks, more
-            let pos = match placed_on.kind() {
-                BlockKind::Grass | BlockKind::TallGrass | BlockKind::Water | BlockKind::Lava => {
-                    packet.location
+            // Decide whether the player should place a block or interact with the block they are targeting
+            // TODO: Maybe player shifting may need to be taken into account (shift click on interactable block)
+            match target_block.kind() {
```

Comment:
It would be cool if feather-data was able to serialize whether the block can be interacted with, but I'm not sure how feasible that is.

Reply:
Not possible, since vanilla doesn't give us that information.

What you can do is manually write the mappings in `feather-definitions`. Take a look at the `data` directory there; it has RON files which specify e.g. block properties. You can add a boolean property "interactable" there.